vtex-apps/return-app
# Custom return options and extra request comment

#### What is the purpose of this pull request?
- Add the ability to have custom return options, with the option to set custom max days those options will be available in regards of the order date.
- Add the ability for the user to add a custom comment to their return request (max 250 chars)

#### What problem is this solving?
- There was no way for the user to add an extra comment regarding the whole request
- There was no way for the owner to dynamically have refund options that could be turned on/off
- There was no way for an option to be available or not, regarding a difference between the current date and the order date
- If the user doesn't have an address No. the address will not show a literal 'null' string
- When navigating the my account forms, now the page will scroll to top
- Better admin settings spacing

#### How should this be manually tested?
- First, you should access the admin return list or settings, to update the schema.
- (Optional) Add custom options in settings.
- With a store user, make a purchase, force it to 'Invoiced' status; access my account page, go to my returns.
- There you'll see the extra comment field and, if done so, you'll see your custom return options.

#### Disclaimer
All changes are retroactive.
You are not required to add custom return options, if left empty, the default (previous) options will be user. 

#### Screenshots or example usage
<img width="1440" alt="settings_options" src="https://user-images.githubusercontent.com/50715158/122242383-2f239780-cebb-11eb-8a01-42b7ac5f64b1.png">
<img width="1440" alt="request_comment_options" src="https://user-images.githubusercontent.com/50715158/122243163-cf79bc00-cebb-11eb-8726-f759024e43c0.png">
<img width="705" alt="return_list" src="https://user-images.githubusercontent.com/50715158/122242395-32b71e80-cebb-11eb-91f9-1e48c431f44b.png">
